src/Stratis.Bitcoin.Features.Interop/InteropFeature.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Builder;
+using Stratis.Bitcoin.Builder.Feature;
+using Stratis.Bitcoin.Configuration.Logging;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.Features.Interop.EthereumClient;
+using Stratis.Bitcoin.Features.PoA;
+using Stratis.Bitcoin.P2P.Peer;
+
+namespace Stratis.Bitcoin.Features.Interop
+{
+    public sealed class InteropFeature : FullNodeFeature
+    {
+        private readonly ILoggerFactory loggerFactory;
+        
+        private readonly Network network;
+
+        private readonly IFederationManager federationManager;
+
+        private readonly IConnectionManager connectionManager;
+        
+        private readonly InteropPoller interopPoller;
+        
+        private readonly IInteropTransactionManager interopTransactionManager;
+        
+        private readonly IEthereumClientBase ethereumClientBase;
+
+        public InteropFeature(ILoggerFactory loggerFactory, 
+            Network network, 
+            IFederationManager federationManager,
+            IConnectionManager connectionManager,
+            InteropPoller interopPoller,
+            IInteropTransactionManager interopTransactionManager, 
+            IEthereumClientBase ethereumClientBase)
+        {
+            this.loggerFactory = loggerFactory;
+            this.network = network;
+            this.federationManager = federationManager;
+            this.connectionManager = connectionManager;
+            this.interopPoller = interopPoller;
+            this.interopTransactionManager = interopTransactionManager;
+            this.ethereumClientBase = ethereumClientBase;
+        }
+
+        public override Task InitializeAsync()
+        {
+            this.interopPoller?.Initialize();
```

Comment:
How do we guarantee that this is called after the federation feature has initialized? As we rely on things like "IsFederationMember" etc.

Reply:
Hmm, the interoperability feature should initialize well after the federation manager, is it necessary to explicitly check that it has completed before using IsFederationMember?

Reply:
I think it's ok. We just need to ensure that initialization sequence is correct